core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,110 @@
-          function(response) {
-            var playthroughBackendDict = response.data;
-            return PlaythroughObjectFactory.createFromBackendDict(
-              playthroughBackendDict);
-          });
-      },
-      resolveIssue: function(issueToResolve, expId, expVersion) {
-        return $http.post(getFullResolveIssueUrl(expId), {
-          exp_issue_dict: issueToResolve.toBackendDict(),
-          exp_version: expVersion
-        }).then(function() {
-          var issueIndex = cachedIssues !== null ?
-            cachedIssues.findIndex(function(issue) {
-              return angular.equals(issue, issueToResolve);
-            }) : -1;
-          if (issueIndex === -1) {
-            var invalidIssueError = new Error(
-              'An issue which was not fetched from the backend has been ' +
-              'resolved');
-            return $q.reject(invalidIssueError);
-          } else {
-            cachedIssues.splice(issueIndex, 1);
-          }
+  private getFullIssuesUrl(explorationId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_ISSUES_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  private getFullPlaythroughUrl(
+      explorationId: string, playthroughId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_PLAYTHROUGH_URL, {
+        exploration_id: explorationId,
+        playthrough_id: playthroughId
+      });
+  }
+
+  private getFullResolveIssueUrl(explorationId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.RESOLVE_ISSUE_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  // TODO(#7165): This has been marked any since marking explorationVersion
+  // to number throws an error. "Type 'number' is not assignable to type
+  // 'string | string[]'" and if this is marked to string it throws an
+  // error "TS2345: Argument of type '1' is not assignable to parameter
+  // of type 'string'".
+  fetchIssues(explorationId: string, explorationVersion: any): Promise<string> {
+    if (this.cachedIssues !== null) {
+      return new Promise((resolve) => resolve(this.cachedIssues));
+    } else {
+      return this.httpClient.get(
+        this.getFullIssuesUrl(explorationId), {
+          params: { exp_version: explorationVersion }, observe: 'response'
+        }).toPromise().then(
+        (response: HttpResponse<Array<PlaythroughIssue>>) => {
+          let unresolvedIssueBackendDicts = response.body;
+          this.cachedIssues = unresolvedIssueBackendDicts.map(
+            this.playthroughIssueObjectFactory.createFromBackendDict);
+          return this.cachedIssues;
         });
-      },
-    };
-  }]);
+    }
+  }
+
+  fetchPlaythrough(
+      explorationId: string, playthroughId: string): Promise<PlaythroughIssue> {
+    return this.httpClient.get(
+      this.getFullPlaythroughUrl(
```

Comment:
Please prefer keeping the opening of config-objects on the line of the URL, for example, please change this to:
```typescript
return this.httpClient.get(
  this.getFullPlaythroughUrl(explorationId, playthroughId), {
    observe: 'response'
  })).toPromise().then(...);
```